File: 10250/10250.py
# **************************************************************************** #
#                                                                              #
#                                                         :::      ::::::::    #
#    10250.py                                           :+:      :+:    :+:    #
#                                                     +:+ +:+         +:+      #
#    By: jihoolee <jihoolee@student.42.fr>          +#+  +:+       +#+         #
#                                                 +#+#+#+#+#+   +#+            #
#    Created: 2021/09/28 15:09:49 by jihoolee          #+#    #+#              #
#    Updated: 2021/09/28 15:54:13 by jihoolee         ###   ########.fr        #
#                                                                              #
# **************************************************************************** #

def get_suitable_room(max_floor, max_room, guest_num):
	floor = guest_num % max_floor
	room = guest_num // max_floor + 1
	if (floor == 0):
		floor = max_floor
		room -= 1
	return (floor * 100 + room)


cases = int(input())
inputs_lst = [tuple(map(int, input().split())) for _ in range(cases)]

for case in inputs_lst:
	print(get_suitable_room(*case))

File: 10250/10250.cpp
/* ************************************************************************** */
/*                                                                            */
/*                                                        :::      ::::::::   */
/*   10250.cpp                                          :+:      :+:    :+:   */
/*                                                    +:+ +:+         +:+     */
/*   By: jihoolee <jihoolee@student.42.fr>          +#+  +:+       +#+        */
/*                                                +#+#+#+#+#+   +#+           */
/*   Created: 2021/09/28 14:43:20 by jihoolee          #+#    #+#             */
/*   Updated: 2021/09/28 15:09:13 by jihoolee         ###   ########.fr       */
/*                                                                            */
/* ************************************************************************** */

#include <iostream>
#include <vector>

using namespace std;

int	get_suitable_room(int max_floor, int max_roomnum, int guestnum)
{
	int	floor, room;

	floor = guestnum % max_floor;
	room = guestnum / max_floor + 1;
	if (floor == 0)
	{
		floor = max_floor;
		room--;
	}
	return (floor * 100 + room);
}

int	main(void)
{
	int			cases;
	cin >> cases;
	vector<int>	max_floor(cases),
				max_roomnum(cases),
				guestnum(cases);

	for (int i = 0; i < cases; i++)
		cin >> max_floor[i] >> max_roomnum[i] >> guestnum[i];
	for (int i = 0; i < cases; i++)
		cout <<
			get_suitable_room(max_floor[i], max_roomnum[i], guestnum[i])
			<< endl;;
	return (0);
}
